pkg/reconciler/delivery/delivery.go

```diff
@@ -0,0 +1,156 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package delivery
+
+import (
+	"context"
+	"fmt"
+
+	clientset "knative.dev/serving/pkg/client/clientset/versioned"
+	configurationreconciler "knative.dev/serving/pkg/client/injection/reconciler/serving/v1/configuration"
+
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/ptr"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	listers "knative.dev/serving/pkg/client/listers/serving/v1"
+)
+
+const (
+	// ReconcilerName is the name of the reconciler
+	ReconcilerName = "Delivery"
+	// KCDNamespace is the namespace of this project (KCD = Knative Continuous Delivery)
+	KCDNamespace = "knative-serving"
+	// KCDName is the name of this project
+	KCDName = "knative-continuous-delivery"
+)
+
+// Reconciler implements controller.Reconciler
+type Reconciler struct {
+	client clientset.Interface
+
+	configurationLister listers.ConfigurationLister
+	revisionLister      listers.RevisionLister 
+	routeLister         listers.RouteLister 
+}
+
+// Check that our Reconciler implements ksvcreconciler.Interface
+var _ configurationreconciler.Interface = (*Reconciler)(nil)
+var (
+	newPercent int64 = 50
+	oldPercent int64 = 50
+)
+
+// ReconcileKind is a very simple proof-of-concept reconciliation method
+// Assumes that there is one existing Revision and one new Revision
+// when the new Revision arrives, split traffic between 2 Revisions
+// according to newPercent/oldPercent
+func (c *Reconciler) ReconcileKind(ctx context.Context, cfg *v1.Configuration) pkgreconciler.Event {
+	// ignore changes triggered by continuous-delivery itself
+	if shouldSkipConfig(cfg) {
+		return nil
+	}
+
+	// wait for latest created Revision to be ready
+	if !configReady(cfg) {
+		return nil
+	}
+
+	return c.updateRoute(ctx, cfg)
+}
+
+// shouldSkipConfig determines if we should do a no-op because the reconciler is triggered
+// by changes in KCD itself
+func shouldSkipConfig(cfg *v1.Configuration) bool {
+	return cfg.Namespace == KCDNamespace && cfg.Name == KCDName
+}
+
+// configReady determines if the given Configuration's latest created Revision is ready
+func configReady(cfg *v1.Configuration) bool {
+	latestReady := cfg.Status.LatestReadyRevisionName
+	latestCreated := cfg.Status.LatestCreatedRevisionName
+	return latestReady == latestCreated
+}
+
+// updateRoute figures out if the Route object needs any update and updates it as needed
+func (c *Reconciler) updateRoute(ctx context.Context, cfg *v1.Configuration) error {
+	logger := logging.FromContext(ctx)
+
+	// find the Route object
+	r, err := c.routeLister.Routes(cfg.Namespace).Get(cfg.Name)
+	if err != nil {
+		return err
+	}
+	route := r.DeepCopy()
+	latestReady := cfg.Status.LatestReadyRevisionName
+
+	// update the Route if necessary
+	if routeStatusUptodate(route, latestReady) {
```

Comment:
nit: boolean functions should read like a question, something like `isRouteStatusUpToDate` or `shouldUpdateRouteStatus`?